git_creation_and_tracking/git_creation_and_tracking.md

```diff
@@ -0,0 +1,829 @@
+
+```
+$ mkdir results
+$ touch a.dat b.dat c.dat results/a.out results/b.out
+```
+
+The `touch` command opens, saves, and closes the files without changing them. Since they didn't previously exist, `touch` is also creating them. Now if we check the status of our repository we see:
+
+```
+$ git status
+
+
+On branch main
+Untracked files:
+ (use "git add <file>..." to include in what will be committed)
+
+	a.dat
+	b.dat
+	c.dat
+	results/
+
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+In our scenario, these are files we don't want to track.  Putting these files under version control would be a waste of disk space. What's worse, having them all listed could distract us from changes that actually matter, so let's tell Git to ignore them.
+
+We do this by creating a file in the root directory of our project called `.gitignore`.   That dot at the beginning is intentional!  The file must be named this exactly for `.gitignore` to work properly.  Each line in this file is a rule for a type of file Git should ignore. Use your favorite text editor to make `.gitignore` contain the two lines:
+
+```
+*.dat
+results/
+```
+
+These patterns tell Git to ignore any file whose name ends in `.dat` and everything in the `results` directory. (If any of these files were already being tracked, Git would continue to track them.)
+
+<div class = "learnmore">
+There are many common file types that people frequently ask Git not to track. For example Mac users usually want Git to ignore `.DS_store`, which is an invisible file that gets created by opening a file in Finder. You can even start with [this collection of common configurations](https://gist.github.com/octocat/9257657) for `.gitignore`.
+</div>
+
+Once we have created the `.gitignore` file, the output of `git status` is much cleaner:
+
+```
+$ git status
+
+On branch main
+Untracked files:
+ (use "git add <file>..." to include in what will be committed)
+
+	.gitignore
+
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+The only thing Git notices now is the newly-created `.gitignore` file. You might think we wouldn't want to track it, but if we share the repository, everyone we're sharing with will probably want to ignore the same things that we are ignoring.
+Let's add and commit `.gitignore`:
+
+```
+$ git add .gitignore
+$ git commit -m "Ignore data files and the results folder."
+$ git status
+
+On branch main
+
+nothing to commit, working directory clean
+```
+
+As a bonus, using `.gitignore` helps us avoid accidentally adding files to the repository that we don't want to track:
+
+```
+$ git add a.dat
+
+The following paths are ignored by one of your .gitignore files:
+
+ a.dat
+
+Use -f if you really want to add them.
+```
+
+If we really want to override our ignore settings,
+we can use `git add -f` to force Git to add something. For example,
+`git add -f a.dat`.
+We can also always see the status of ignored files if we want:
```

Comment:
Add an additional blank line space prior to this sentence to separate from the override ignore example.